core/src/test/java/org/projectnessie/cel/extension/StringsTest.java

```diff
@@ -0,0 +1,349 @@
+/*
+ * Copyright (C) 2022 The Authors of CEL-Java
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.cel.extension;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import org.junit.jupiter.api.Test;
+import org.projectnessie.cel.Env;
+import org.projectnessie.cel.Library;
+import org.projectnessie.cel.Program;
+import org.projectnessie.cel.common.types.pb.ProtoTypeRegistry;
+
+public class StringsTest {
```

Comment:
Thanks for adding all the tests!

Can you change the repeated calls to testExpression() to parameterized tests like this:

```
  @ParameterizedTest
  @MethodSource("testCharAt")
  public void testCharAt(TestData testData) {
    testExpression(testData);
  }

  static Stream<Arguments> testCharAt() {
    return Stream.of(
      new TestData(...),
      new TestData(...)
    );
  }
```

This will run every test case, also those after a failed one.

As an alternative, you can also wrap all `testExpression` invocations in a `assertAll` like this:
```
    assertAll(
            () -> testExpression(...),
            () -> testExpression(...),
            );
```

I'd be fine with either.

Reply:
Thanks, this is very useful to simplify the test code